src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -1367,6 +1367,45 @@ public void ThrowExceptionForCustomImplementationOfIHostApplicationLifetime()
             }
         }
 
+        /// <summary>
+        /// Tests that when a BackgroundService is canceled when stopping the host,
+        /// no error is logged.
+        /// </summary>
+        [Fact]
+        public async Task HostNoErrorWhenServiceIsCanceledAsPartOfStop()
+        {
+            using TestEventListener listener = new TestEventListener();
```

Comment:
does this work? How do you make sure its the right instance ?

Reply:
Oh yeah, it works. But we only use it in 2 tests (this one and the one above it). Since they are in the same class, they don't run in parallel in xunit.

Is there a better pattern for "let me see what got logged"? Make my own fake logger?

Reply:
Yep

Reply:
Fixed